libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1556,29 +1593,51 @@ bool PbftManager::pushPbftBlock_(SyncBlock &sync_block, vec_blk_t &dag_blocks_or
   db_->commitWriteBatch(batch);
 
   // Set DAG blocks period
-  auto const &anchor_hash = sync_block.pbft_blk->getPivotDagBlockHash();
-  dag_mgr_->setDagBlockOrder(anchor_hash, pbft_period, dag_blocks_order);
+  // TODO: could be passed only SyncBlock -> a lot of refactoring in tests
+  dag_mgr_->setDagBlockOrder(pbft_block->getPivotDagBlockHash(), pbft_block->getPeriod(),
+                             sync_block.getDagBlocksHashes());
 
-  trx_mgr_->updateFinalizedTransactionsStatus(sync_block);
+  trx_mgr_->updateFinalizedTransactionsStatus(sync_block.getTransactionsHashes());
 
   // update PBFT chain size
   pbft_chain_->updatePbftChain(pbft_block_hash);
 
   last_cert_voted_value_ = NULL_BLOCK_HASH;
 
-  LOG(log_nf_) << node_addr_ << " successful push unexecuted PBFT block " << pbft_block_hash << " in period "
-               << pbft_period << " into chain! In round " << getPbftRound();
+  LOG(log_nf_) << "Pushed new PBFT block " << pbft_block_hash << " into chain. Period: " << pbft_block->getPeriod()
+               << ", round: " << getPbftRound();
+
+  // Transform cert votes into the set of hashes
+  const auto &cert_votes = sync_block.getCertVotes();
+  std::unordered_set<vote_hash_t> cert_votes_hashes;
+  cert_votes_hashes.reserve(cert_votes.size());
+  for (const auto &vote : cert_votes) {
+    cert_votes_hashes.insert(vote->getHash());
+  }
 
-  finalize_(*sync_block.pbft_blk, move(dag_blocks_order));
+  finalize_(std::move(sync_block));
 
   // Reset proposed PBFT block hash to False for next pbft block proposal
   proposed_block_hash_ = std::make_pair(NULL_BLOCK_HASH, false);
   db_->savePbftMgrStatus(PbftMgrStatus::ExecutedBlock, true);
   executed_pbft_block_ = true;
+
+  // Reset 2t+1 cert votes in rewards votes(votes that should be rewarded in the next pbft period)
+  rewards_votes_->resetVotes(std::move(cert_votes_hashes), pbft_block_hash);
```

Comment:
There are only 2t+1 cert votes because 2t+1 cert votes are enough for validate PBFT blocks syncing. But there are may more cert votes in verified queue. If you are going to rewards all cert votes, you may need to write a new function to return all cert votes in verified queue.